apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/DirectedReadTests.cs

```diff
@@ -0,0 +1,456 @@
+        var session = PooledSession
+            .FromSessionName(sessionPool, s_sessionName)
+            .WithTransaction(s_transactionId, s_readOnly, singleUseTransaction);
+
+        await session.ExecuteSqlAsync(new ExecuteSqlRequest(), callSettings: null);
+        Assert.Equal(IncludeDirectedReadOptions, grpcClient.LastExecuteSqlRequest.DirectedReadOptions);
+    }
+
+    [Theory]
+    [CombinatorialData]
+    public async Task PooledSession_SetsOptionsFromRequest_ExecuteSqlAsync(bool singleUseTransaction)
+    {
+        var grpcClient = new FakeGrpcSpannerClient();
+        var spannerClient = new SpannerClientImpl(grpcClient, settings: null, logger: null);
+        var sessionPool = new FakeSessionPool(spannerClient);
+        var session = PooledSession
+            .FromSessionName(sessionPool, s_sessionName)
+            .WithTransaction(s_transactionId, s_readOnly, singleUseTransaction);
+
+        await session.ExecuteSqlAsync(new ExecuteSqlRequest
+        {
+            DirectedReadOptions = IncludeDirectedReadOptions
+        }, callSettings: null);
+
+        Assert.Equal(IncludeDirectedReadOptions, grpcClient.LastExecuteSqlRequest.DirectedReadOptions);
+    }
+
+    [Theory]
+    [CombinatorialData]
+    public async Task PooledSession_RequestOptionsTakePrecedenceOverClientOptions_ExecuteSqlAsync(bool singleUseTransaction)
+    {
+        var grpcClient = new FakeGrpcSpannerClient();
+
+        var spannerClient = new SpannerClientImpl(grpcClient, new SpannerSettings
+        {
+            DirectedReadOptions = ExcludeDirectedReadOptions
+        }, logger: null);
+
+        var sessionPool = new FakeSessionPool(spannerClient);
+        var session = PooledSession
+            .FromSessionName(sessionPool, s_sessionName)
+            .WithTransaction(s_transactionId, s_readOnly, singleUseTransaction);
+
+        await session.ExecuteSqlAsync(new ExecuteSqlRequest
+        {
+            DirectedReadOptions = IncludeDirectedReadOptions
+        }, callSettings: null);
+
+        Assert.Equal(IncludeDirectedReadOptions, grpcClient.LastExecuteSqlRequest.DirectedReadOptions);
+    }
+
+    [Theory]
+    [MemberData(nameof(NonReadOnlyTransactionOptions))]
+    public async Task PooledSession_IgnoresOptionsFromClient_ExecuteSqlAsync(TransactionOptions options)
+    {
+        var grpcClient = new FakeGrpcSpannerClient();
+
+        var spannerClient = new SpannerClientImpl(grpcClient, new SpannerSettings
+        {
+            DirectedReadOptions = IncludeDirectedReadOptions
+        }, logger: null);
+
+        var sessionPool = new FakeSessionPool(spannerClient);
+        var session = PooledSession
+            .FromSessionName(sessionPool, s_sessionName)
+            // Only read-only transaction can be single use.
+            .WithTransaction(s_transactionId, options, singleUseTransaction: false);
+
+        await session.ExecuteSqlAsync(new ExecuteSqlRequest(), callSettings: null);
```

Comment:
I think it would be good to have a comment somewhere to make it clear this was a decision rather than just an omission, but that's all. Maybe even a comment in both the implementation and the test? I'm not fussy - I'm just thinking of "future me who has forgotten this conversation" :)

Reply:
Yep, makes sense of course. I'm adding the comments in implementation and tests.